runelite-client/src/main/java/net/runelite/client/plugins/puzzlesolver/solver/pathfinding/IDAStarMM.java

```diff
@@ -0,0 +1,787 @@
+							swap(locBlank.getX(), locBlank.getY(), locSwap.getX(), locSwap.getY());
+						}
+					}
+				}
+			}
+			else
+			{
+				//Swap towards locTarget
+				if (rowMode)
+				{
+					alignTargetX(valTarget, x, y);
+					swapUpRow(valTarget, x, y);
+				}
+				else
+				{
+					alignTargetY(valTarget, x, y);
+					swapLeftColumn(valTarget, x, y);
+				}
+			}
+		}
+	}
+
+	private void alignTargetX(int valTarget, int x, int y)
+	{
+		Point locVal = findPiece(valTarget);
+
+		//Check if same column
+		if (locVal.getX() == x)
+		{
+			return;
+		}
+
+		//1 = right, -1 = left
+		int direction = Integer.signum(x - locVal.getX());
+
+		while (locVal.getX() != x)
+		{
+			locVal = findPiece(valTarget);
+			Point locBlank = findPiece(BLANK_TILE_VALUE);
+
+			//Check if aligned
+			if (x - locVal.getX() == 0)
+			{
+				break;
+			}
+
+			if (locVal.getX() == locBlank.getX())
+			{
+				int diff = locBlank.getY() - locVal.getY();
+				if (diff == 1)
+				{
+					//Below
+					Point loc1 = new Point(locBlank.getX() + direction, locBlank.getY());
+					Point loc2 = new Point(loc1.getX(), loc1.getY() - 1);
+
+					swap(locBlank.getX(), locBlank.getY(), loc1.getX(), loc1.getY());
+					swap(loc1.getX(), loc1.getY(), loc2.getX(), loc2.getY());
+					swap(loc2.getX(), loc2.getY(), locVal.getX(), locVal.getY());
+				}
+				else if (diff == -1)
+				{
+					//Above
+					swap(locBlank.getX(), locBlank.getY(), locVal.getX(), locVal.getY());
+				}
+			}
+			else if (locVal.getY() == locBlank.getY())
+			{
+				int diff = locBlank.getX() - locVal.getX();
+				if (diff == 1)
+				{
+					//Right
+					if (direction == 1)
+					{
+						swap(locVal.getX(), locVal.getY(), locBlank.getX(), locBlank.getY());
+					}
+					else if (direction == -1)
+					{
+						//Check space
+						if (locVal.getY() == DIMENSION - 1)
+						{
+							//No space below, use upper rotate
+							Point loc1 = new Point(locVal.getX() + 1, locVal.getY() - 1);
+							Point loc2 = new Point(locVal.getX(), locVal.getY() - 1);
+							Point loc3 = new Point(locVal.getX() - 1, locVal.getY() - 1);
+							Point loc4 = new Point(locVal.getX() - 1, locVal.getY());
+
+							swap(locBlank.getX(), locBlank.getY(), loc1.getX(), loc1.getY());
+							swap(loc1.getX(), loc1.getY(), loc2.getX(), loc2.getY());
+							swap(loc2.getX(), loc2.getY(), loc3.getX(), loc3.getY());
+							swap(loc3.getX(), loc3.getY(), loc4.getX(), loc4.getY());
+							swap(loc4.getX(), loc4.getY(), locVal.getX(), locVal.getY());
+						}
+						else
+						{
+							//Space below, use lower rotate
+							Point loc1 = new Point(locVal.getX() + 1, locVal.getY() + 1);
+							Point loc2 = new Point(locVal.getX(), locVal.getY() + 1);
+							Point loc3 = new Point(locVal.getX() - 1, locVal.getY() + 1);
+							Point loc4 = new Point(locVal.getX() - 1, locVal.getY());
+
+							swap(locBlank.getX(), locBlank.getY(), loc1.getX(), loc1.getY());
+							swap(loc1.getX(), loc1.getY(), loc2.getX(), loc2.getY());
```

Comment:
Oh, I see now. The points are used in more than one swap.. Didn't see that, sorry.
I think the best thing in that case would be to have the swap in this class take in two Points like you had before, and have the PuzzleState swap take in integers like it is now.